packages/features/flags/features.repository.ts

```diff
@@ -70,62 +121,83 @@ export class FeaturesRepository implements IFeaturesRepository {
    */
   private async checkIfUserBelongsToTeamWithFeature(userId: number, slug: string) {
     try {
-      const user = await db.user.findUnique({
-        where: {
-          id: userId,
-          teams: {
-            some: {
-              team: {
-                features: {
-                  some: {
-                    featureId: slug,
-                  },
-                },
-              },
-            },
-          },
-        },
-        select: { id: true },
-      });
-      if (user) return true;
-      return false;
+      const query = Prisma.sql`
+        WITH RECURSIVE TeamHierarchy AS (
+          -- Start with teams the user belongs to
+          SELECT t.id, t."parentId"
+          FROM "Team" t
+          INNER JOIN "Membership" m ON m."teamId" = t.id
+          WHERE m."userId" = ${userId}
+
+          UNION ALL
+
+          -- Recursively get parent teams
+          SELECT t.id, t."parentId"
+          FROM "Team" t
+          INNER JOIN TeamHierarchy th ON t.id = th."parentId"
+        )
+        -- Check if any team in the hierarchy has the feature
+        SELECT 1
+        FROM "TeamFeatures" tf
+        WHERE tf."featureId" = ${slug}
+        AND tf."teamId" IN (SELECT id FROM TeamHierarchy)
+        LIMIT 1;
```

Comment:
Could this be distinct 

```suggestion
        SELECT DISTINCT 1
        FROM "TeamFeatures" tf
        WHERE tf."featureId" = ${slug}
        AND tf."teamId" IN (SELECT id FROM TeamHierarchy)
        LIMIT 1;
```

Reply:
I think TeamHierarchy can have a teamId and parentId that overlap? Maybe a distinct helps here - havent tested this but could work